src/main/java/org/cubedb/core/CubeImpl.java

```diff
@@ -295,8 +291,12 @@ public void deletePartition(String partitionName) {
 	@Override
 	public void save(String saveFileName) throws IOException {
 		Kryo kryo = CubeUtils.getKryoWithRegistrations();
-		OutputStream zip = new BufferedOutputStream(new GZIPOutputStream(new FileOutputStream(saveFileName)));
-		Output output = new Output(zip);
+		OutputStream stream;
+		if(saveFileName.endsWith(".gz"))
+			stream = new GZIPOutputStream(new FileOutputStream(saveFileName));
+		else
```

Comment:
if follows from the code that you always save the code with the "*.snappy" extention. How about forcing it here by explicitly checking the extension..? And reporting an error - or at least warning - if the extension is not recognized?